server/src/main/java/io/crate/execution/dml/Indexer.java

```diff
@@ -584,10 +584,9 @@ public void updateTargets(Function<ColumnIdent, Reference> getRef) {
                 // the reference may be invalid
                 Reference newRef = getRef.apply(oldRef.column());
                 if (newRef == null) {
-                    // column was dropped or new column is invalid
-                    it.remove();
-                    valueIndexers.remove(idx);
-                    // don't increase idx, since we removed the current element
+                    // column can be of an undetermined type, e.g. `[]` (array with undefined inner type)
```

Comment:
I've run integ tests with also dropping an existing column and then re-adding it dynamically, and this still works. Is there some other case that I might be missing?

(`oldRef.isDropped()` was never `true` here, during my tests)

Reply:
Random guess, couldn't it be to catch any concurrent drop columns?

Reply:
It's already atomic, AFAIK and as I tested, so dropping a column won't proceed, while another thread is executing `updateTargets()`.